compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java

```diff
@@ -4820,6 +4850,14 @@ private STNode createMissingSimpleVarDecl(STNode metadata, STNode publicQualifie
                 emptyNode, semicolon);
     }
 
+    private STNode createMissingWhereClause() {
+        STNode whereKeyword = SyntaxErrors.createMissingTokenWithDiagnostics(SyntaxKind.WHERE_KEYWORD,
+                DiagnosticErrorCode.ERROR_MISSING_WHERE_KEYWORD);
+        STNode expression = SyntaxErrors.createMissingTokenWithDiagnostics(SyntaxKind.BOOLEAN_LITERAL,
```

Comment:
BOOLEAN_LITERAL is not a token. It is a node containing literal token TRUE/FLASE_KEYWORD token. 
Anyway in other place when we want something for the missing expression. We go for the missing simple name reference. For the consistency let's create the here as well.

Reply:
```java
STNode missingIdentifier = SyntaxErrors.createMissingTokenWithDiagnostics(SyntaxKind.IDENTIFIER_TOKEN,DiagnosticErrorCode.ERROR_MISSING_EXPRESSION);
STNode missingExpr = STNodeFactory.createSimpleNameReferenceNode(missingIdentifier);
```
Let's use this.

Reply:
Updated.